Sources/Plasma/CoreLib/plFileSystem.cpp

```diff
@@ -460,7 +469,36 @@ plFileName plFileSystem::GetUserDataPath()
 
         _userData = plFileName::Join(ST::string::from_wchar(path), plProduct::LongName());
 #else
-        _userData = plFileName::Join(getenv("HOME"), "." + plProduct::LongName());
+        ST::string dir;
+#  if HS_BUILD_FOR_APPLE
+        char home[256] = "";
+        if (__builtin_available(macOS 10.12, *)) {
+            sysdir_search_path_enumeration_state state;
+            state = sysdir_start_search_path_enumeration(SYSDIR_DIRECTORY_APPLICATION_SUPPORT, SYSDIR_DOMAIN_MASK_USER);
+            state = sysdir_get_next_search_path_enumeration(state, home);
+        } else {
+            #pragma clang diagnostic push
+            #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+            NSSearchPathEnumerationState state;
+            state = NSStartSearchPathEnumeration(NSApplicationSupportDirectory, NSUserDomainMask);
+            state = NSGetNextSearchPathEnumeration(state, home);
+            #pragma clang diagnostic pop
+        }
+        dir = home;
+#  else
+        dir = "~" PATH_SEPARATOR_STR ".config";
```

Comment:
To fully follow the XDG spec, this should use the value of the `XDG_CONFIG_HOME` variable, and only if that variable isn't defined fall back to the default `~/.config`.